File: backend/app/database/models/business.py
# app/models/business.py
from typing import List, Optional
from datetime import datetime
from pydantic import BaseModel, Field

class Location(BaseModel):
    type: str = "Point"
    coordinates: List[float]

class Address(BaseModel):
    street: str
    area: str
    city: str
    pincode: str

class Contact(BaseModel):
    phone: str
    whatsapp: Optional[str] = None
    email: Optional[str] = None

class Business(BaseModel):
    id: Optional[str] = Field(default=None, alias="_id")
    businessName: str
    shortName: str
    category: str
    subCategory: str
    location: Location
    address: Address
    contact: Contact
    tags: List[str]
    createdAt: Optional[datetime] = None
    updatedAt: Optional[datetime] = None

    class Config:
        populate_by_name = True

File: frontend/src/types/business.ts
export interface Location {
    type: "Point";
    coordinates: [number, number];
  }
  
  export interface Address {
    street: string;
    area: string;
    city: string;
    pincode: string;
  }
  
  export interface Contact {
    phone: string;
    whatsapp?: string;
    email?: string;
  }
  
  export interface Business {
    businessName: string;
    shortName: string;
    category: string;
    subCategory: string;
    location: Location;
    address: Address;
    contact: Contact;
    tags: string[];
  }